plugin.py

```diff
@@ -93,6 +107,51 @@ def install_or_update(cls) -> None:
             shutil.rmtree(cls.basedir(), ignore_errors=True)
             raise
 
+    def on_pre_server_command(self, command: Mapping[str, Any], done_callback: Callable[[], None]) -> bool:
+
+        cargo_commands = []
+        for c in command["arguments"]:
+            if c["kind"] == "cargo":
+                cargo_commands.append(c)
+
+        if len(cargo_commands) == 0:
+            return False
+
+        window = sublime.active_window()
+        if window is None:
+            return False
+        view = window.active_view()
+        if not view:
+            return False
+        if not Terminus:
+            sublime.error_message(
+                'Cannot run executable. You need to install the "Terminus" package and then restart Sublime Text')
+            done_callback()
+            return True
+        if not shutil.which("cargo"):
+            sublime.error_message('Cannot find "cargo" on path.')
+            return False
+        main_cargo_path = '"{}"'.format(shutil.which("cargo"))
+        for output in cargo_commands:
+            if output["args"]["overrideCargo"]:
+                cargo_path = output["args"]["overrideCargo"]
+            else:
+                cargo_path = main_cargo_path
+            command_to_run = [cargo_path] + output["args"]["cargoArgs"] + \
+                output["args"]["cargoExtraArgs"]
+            cmd = " ".join(command_to_run)
+            args = {
+                "title": output["label"],
+                "shell_cmd": cmd,
+                "cwd": output["args"]["workspaceRoot"],
+                "auto_close": get_setting(view, "terminus_auto_close", False)
+            }
+            if get_setting(view, "terminus_use_panel", False):
+                args["panel_name"] = output["label"]
+            window.run_command("terminus_open", args)
```

Comment:
Shouldn't this be _outside_ the for-loop above? Are we starting shells for every cargo_command?

Reply:
Lets say a codelens has multiple runnables,  this spawns a shell for all of the elements in the Runnables array. Very rare but exists!